Claude/code revisión

litio2001 opened the pull request:
✅ Notificación Premium mejorada - Mensaje visual claro al inicio
    ✅ Comando de ayuda - Lista todos los comandos disponibles
    ✅ Estadísticas de descarga - Contador de archivos, tamaño total, velocidad
    ✅ Sistema de reintentos - Reintenta descargas fallidas automáticamente
    ✅ Comando config - Muestra configuración actual
    ✅ Velocidad de descarga - Muestra velocidad en tiempo real
    ✅ Mejor logging - Logs más detallados y útiles
    ✅ Filtros de archivo - Filtrar por extensión
    ✅ Comando pause/resume - Pausar y reanudar cola

litio2001 committed:
Fix clean command output type

litio2001 committed:
Merge pull request #1 from litio2001/codex/encontrar-y-corregir-fallo-en-c-digo

Corregir el tipo de salida del comando limpio

litio2001 committed:
update

litio2001 committed:
update_premium

litio2001 committed:
update_premium

litio2001 committed:
update

litio2001 committed:
update

litio2001 committed:
update

litio2001 committed:
update

litio2001 committed:
update

litio2001 committed:
update

litio2001 committed:
update

litio2001 committed:
update

litio2001 committed:
update

claude committed:
feat: Release v2.0-Premium-Enhanced with major UX improvements

🌟 Major New Features:
- Visual Premium Status Notification: Welcome message in Telegram channel
  showing Premium/Standard status, optimizations, and configuration
- Real-time Statistics: Session stats with downloads, speed, data transferred
- Interactive Commands: 9 new commands (help, config, stats, pause, resume)
- Speed Monitoring: Real-time download speed with ETA calculation
- Auto-retry System: Automatic retry up to 3 times with exponential backoff
- File Filters: Filter downloads by file extension (TELEGRAM_DAEMON_FILE_FILTER)

💬 Enhanced User Experience:
- Beautiful formatted messages with clear sections and emojis
- Detailed progress updates showing speed, ETA, and formatted sizes
- Contextual error messages with helpful solutions
- Premium upgrade suggestions for standard users
- Helper functions for formatting bytes, speed, and time

🔧 Technical Improvements:
- Better code organization with utility functions
- Enhanced statistics tracking (speeds, largest file, success rate)
- Improved error handling with specific suggestions
- Queue pause/resume functionality
- Session uptime tracking

📝 New Commands:
- help: Show all available commands
- config: Display current configuration
- stats: View detailed session statistics
- pause/resume: Control download queue processing

🐛 Bug Fixes:
- Fixed global variable declarations
- Improved download progress tracking
- Better handling of temporary files
- Enhanced duplicate file detection

📚 Documentation:
- Updated README with comprehensive v2.0 documentation
- Added feature comparison table
- Enhanced Docker setup instructions
- Detailed startup message example

claude committed:
fix: Apply Copilot security and thread-safety improvements (v2.0.1)

🔒 Security Improvements:
- Eliminated shell injection risks by removing shell=True from subprocess calls
- list command: Uses subprocess.run(["ls", "-lh", folder]) instead of shell
- clean command: Uses glob + os.remove() instead of shell rm command

🧵 Thread-Safety Fixes:
- Added asyncio.Lock() for all shared data structures (stats, in_progress)
- Replaced global download_paused variable with asyncio.Event() for proper sync
- Protected all stats updates with async locks to prevent race conditions
- Made pause/resume operations thread-safe using Event.clear()/set()

🐛 Bug Fixes:
- Fixed file extension parsing using os.path.splitext() instead of string split
- Corrected stats['total_downloads'] to increment when download starts, not queued
- Replaced all bare except: with except Exception: for better error handling
- Fixed memory leak: Limited download_speeds to 1000 entries using deque(maxlen)

📏 Code Quality:
- Updated version to semantic versioning: 2.0.1-premium-enhanced
- Defined named constants for magic numbers:
  * SPEED_SAMPLES_FOR_AVERAGE = 10
  * MAX_DOWNLOAD_SPEEDS = 1000
  * MAX_RETRIES = 3
  * RETRY_DELAY_BASE = 5
- Added comment clarifying linear backoff strategy (not exponential)
- Used constants throughout worker retry logic

📚 Documentation:
- Added comprehensive Troubleshooting section to README
- Clarified file filter format (extensions without dots)
- Updated docker-compose.yml with extension format comment
- Added v2.0.1 changelog with all improvements

Addresses: 25 issues identified in Copilot code review

alfem commented on telegram-download-daemon.py:
```diff
@@ -258,39 +722,154 @@ async def worker():
+                    error_msg = f"❌ **ERROR EN DESCARGA**\n\n"
+                    error_msg += f"📄 **Archivo:** {filename}\n"
+                    error_msg += f"🚨 **Error:** {str(e)}\n"
+                    error_msg += f"🔄 **Reintentos agotados:** {max_retries}\n\n"
```
This is a tool used worldwide, could you keep the messages in english?

claude committed:
i18n: Translate all messages from Spanish to English

Changed all user-facing messages to English for international usage:

**Welcome Message & System:**
- System information labels (Started, CPU Cores, etc.)
- Account status (Premium/Standard Account Detected)
- Active features list
- Premium optimizations description
- Configuration section

**Commands (help, config, stats, status, pause, resume, clean, queue):**
- Command descriptions and help text
- Configuration display (Account, Paths, Options)
- Statistics labels (Uptime, Downloads, Success rate, etc.)
- Status information
- Pause/resume messages
- Clean temporary files messages
- Queue status messages

**File Operations:**
- File filtered messages
- File too large warnings
- Added to queue notifications
- File already exists messages
- Not downloadable error

**Download Process:**
- Starting download message
- Download mode indicator
- Progress indicators (Speed, ETA)
- Retry messages
- Download completed notifications
- Download error messages with suggestions

**README:**
- Updated startup message example from Spanish to English

All emojis and formatting preserved. Messages now in English
for worldwide tool usage as requested by repository owner.

Addresses maintainer feedback: "This is a tool used worldwide,
could you keep the messages in english?"
